Crash on weave tagger

```
panic: runtime error: slice bounds out of range

goroutine 42 [running]:
github.com/weaveworks/scope/probe/overlay.(*Weave).Tag(0xc82012eb80, 0xc8213d5380, 0xc8213d53b0, 0xc8213d53e0, 0xc8213d5410, 0xc8213d5440, 0xc820fc6510, 0xc820fc6540, 0xc820fd66f0, 0xc820fd6840, ...)
        /go/src/github.com/weaveworks/scope/probe/overlay/weave.go:139 +0x85a
github.com/weaveworks/scope/probe.(*Probe).tag(0xc8200a59e0, 0xc8213d5380, 0xc8213d53b0, 0xc8213d53e0, 0xc8213d5410, 0xc8213d5440, 0xc820fc6510, 0xc820fc6540, 0xc820fd66f0, 0xc820fd6840, ...)
        /go/src/github.com/weaveworks/scope/probe/probe.go:160 +0x200
github.com/weaveworks/scope/probe.(*Probe).spyLoop(0xc8200a59e0)
        /go/src/github.com/weaveworks/scope/probe/probe.go:111 +0x184
created by github.com/weaveworks/scope/probe.(*Probe).Start
        /go/src/github.com/weaveworks/scope/probe/probe.go:85 +0x58
```

--- a/probe/kubernetes/reporter.go
+++ b/probe/kubernetes/reporter.go
@@ -1,7 +1,9 @@
 package kubernetes
 
 import (
+	"github.com/weaveworks/scope/probe/docker"
 	"github.com/weaveworks/scope/report"
+
 	"k8s.io/kubernetes/pkg/labels"
 )
 
@@ -66,11 +68,12 @@ func (r *Reporter) podTopology(services []Service) (report.Topology, report.Topo
 		nodeID := report.MakePodNodeID(p.Namespace(), p.Name())
 		pods = pods.AddNode(nodeID, p.GetNode())
 
-		container := report.MakeNodeWith(map[string]string{
-			PodID:     p.ID(),
-			Namespace: p.Namespace(),
-		}).WithParents(report.EmptySets.Add(report.Pod, report.MakeStringSet(nodeID)))
 		for _, containerID := range p.ContainerIDs() {
+			container := report.MakeNodeWith(map[string]string{
+				PodID:              p.ID(),
+				Namespace:          p.Namespace(),
+				docker.ContainerID: containerID,
+			}).WithParents(report.EmptySets.Add(report.Pod, report.MakeStringSet(nodeID)))
 			containers.AddNode(report.MakeContainerNodeID(containerID), container)
 		}
 		return nil

--- a/probe/kubernetes/reporter.go
+++ b/probe/kubernetes/reporter.go
@@ -1,10 +1,10 @@
 package kubernetes
 
 import (
+	"k8s.io/kubernetes/pkg/labels"
+
 	"github.com/weaveworks/scope/probe/docker"
 	"github.com/weaveworks/scope/report"
-
-	"k8s.io/kubernetes/pkg/labels"
 )
 
 // Reporter generate Reports containing Container and ContainerImage topologies

--- a/probe/overlay/weave.go
+++ b/probe/overlay/weave.go
@@ -134,9 +134,15 @@ func (w *Weave) Tag(r report.Report) (report.Report, error) {
 	}
 
 	// Put information from weave ps on the container nodes
+	const maxPrefixSize = 12
 	for id, node := range r.Container.Nodes {
-		prefix, _ := node.Latest.Lookup(docker.ContainerID)
-		prefix = prefix[:12]
+		prefix, ok := node.Latest.Lookup(docker.ContainerID)
+		if !ok {
+			continue
+		}
+		if len(prefix) > maxPrefixSize {
+			prefix = prefix[:maxPrefixSize]
+		}
 		entry, ok := w.psCache[prefix]
 		if !ok {
 			continue

